include/ddc/kernels/splines/matrix_periodic_banded.hpp

```diff
@@ -0,0 +1,192 @@
+                assert(std::fabs(aij) < 1e-20);
+                return;
+            }
+
+            if (d > 0) {
+                m_lambda(i - m_nb, j) = aij;
+            } else {
+                m_lambda(i - m_nb, j - m_nb + m_k + 1) = aij;
+            }
+        } else {
+            Matrix_Corner_Block<ExecSpace>::set_element(i, j, aij);
+        }
+    }
+
+public:
+    void calculate_delta_to_factorize() override
+    {
+        auto delta_proxy = *m_delta;
+        Kokkos::parallel_for(
+                "calculate_delta_to_factorize",
+                Kokkos::MDRangePolicy<
+                        Kokkos::DefaultHostExecutionSpace,
+                        Kokkos::Rank<2>>({0, 0}, {m_k, m_k}),
+                [&](const int i, const int j) {
+                    double val = 0.0;
+                    // Upper diagonals in lambda, lower diagonals in Abm_1_gamma
+                    for (int l = 0; l < i + 1; ++l) {
+                        val += m_lambda(i, l) * m_Abm_1_gamma(l, j);
+                    }
+                    // Lower diagonals in lambda, upper diagonals in Abm_1_gamma
+                    for (int l = i + 1; l < m_k + 1; ++l) {
+                        int l_full = m_nb - 1 - m_k + l;
+                        val += m_lambda(i, l) * m_Abm_1_gamma(l_full, j);
+                    }
+                    auto tmp = delta_proxy.get_element(i, j);
+                    delta_proxy.set_element(i, j, tmp - val);
+                });
+    }
+
+    ddc::DSpan2D_stride solve_lambda_section(
+            ddc::DSpan2D_stride const v,
+            ddc::DView2D_stride const u) const override
+    {
+        auto lambda_device = create_mirror_view_and_copy(ExecSpace(), m_lambda);
+        auto nb_proxy = m_nb;
+        auto k_proxy = m_k;
+        Kokkos::parallel_for(
+                "solve_lambda_section",
+                Kokkos::TeamPolicy<ExecSpace>(v.extent(1), Kokkos::AUTO),
+                KOKKOS_LAMBDA(
+                        const typename Kokkos::TeamPolicy<ExecSpace>::member_type& teamMember) {
+                    const int j = teamMember.league_rank();
+
+
+                    Kokkos::parallel_for(
+                            Kokkos::TeamThreadRange(teamMember, k_proxy),
+                            [&](const int i) {
+                                /// Upper diagonals in lambda
+                                for (int l = 0; l <= i; ++l) {
+                                    Kokkos::atomic_sub(&v(i, j), lambda_device(i, l) * u(l, j));
+                                }
+                                // Lower diagonals in lambda
+                                for (int l = i + 1; l < k_proxy + 1; ++l) {
+                                    Kokkos::atomic_sub(
+                                            &v(i, j),
+                                            lambda_device(i, l) * u(nb_proxy - 1 - k_proxy + l, j));
+                                }
+                            });
+                });
```

Comment:
You seem to have changed your approach of parallelisation compared to `calculate_delta_to_factorize`. The approach in `calculate_delta_to_factorize` seems simpler, do you think the hierarchical parallelisation is worth it ?

Reply:
`calculate_delta_to_factorize` is perfomed once on CPU during factorization. Here we need a performant matrix manipulation performed at each solver call.

Reply:
I agree it needs to be efficient.

How far have you been in the analysis of performance ?
- We need to know if we have coalescent accesses on GPU. Are `u` and `v` really non-contiguous ? if not, left or right ?
- Atomics have a cost, are atomics really necessary here ?

Reply:
1) Yes u and v are really non-contiguous because rhs structure is something like this:

```
uuuuuuuuuu
uuuuuuuuuu
uuuuuuuuuu
vvvvvvvvvvvv
vvvvvvvvvvvv
```

In Left layout.

2) Looking at it again, I don't think atomics are necessary as long as we keep the loops on `l` sequential (but if we use TeamThreads/ThreadVector we can still use a scalar buffer to compute the sum on `l` before doing the substraction).
3) In my original implementation lambda was on the device. So no, I did not look at the performance impact of it since the change.

Reply:
Should I replace sequential for loops on `l` with ThreadVector actually ?

Reply:
No i think this work would require its own PR to make a study of what execution policy better suits this loop.